iina/AppDelegate.swift

```diff
@@ -646,7 +646,7 @@ class AppDelegate: NSObject, NSApplicationDelegate, SPUUpdaterDelegate {
 
     // Instruct any players that are already stopped to start shutting down.
     for player in PlayerCore.playerCores {
-      if player.isStopped && !player.isShutdown {
+      if player.info.state == .stopped || player.info.state == .idle {
```

Comment:
Here `player.info.state == .idle` is used; in other places `player.isIdle` is used. We should use only one of them in all places. 

For me, I don't see a particular advantage of using the computed properties (other than it will be a little bit shorter). And why we only have `isIdle, isPaused, isPlaying, isSeeking`, what about other states? So I suggest remove all the computed properties, and only use `player.info.state == __`. This style will remind us and new programmers explicitly that we are using `PlayerInfo.State` other than some unknown logic to compute the current state.

Reply:
Will remove the computed properties and replace with direct checking of the state.

Reply:
> So I suggest remove all the computed properties, and only use player.info.state == __ 

In many places we need to check against a range of states. To ensure clarity, how about adding to `PlayerState`:

```Swift
  func isAtLeast(_ minStatus: PlayerState) -> Bool {
    return rawValue >= minStatus.rawValue
  }

  func isNotYet(_ status: PlayerState) -> Bool {
    return rawValue < status.rawValue
  }
```
So in many places, you'd have e.g., `guard player.info.state.isNotYet(.stopping) else { return }`.

Reply:
Responding to myself after reading more of the existing PR code...
> So in many places, you'd have e.g., `guard player.info.state.isNotYet(.stopping) else { return }`.

I found that in practice, the `[.playing, .paused, .seeking]` states are sort of off on their own in relation to the other states.

But writing `player.info.state.isAtLeast(.playing)` doesn't seem very intuitive.

But you don't have to minimize the number of states! You could add another state above `.playing`, called ".started". That way you could write `player.info.state.isAtLeast(.started)`. But you'd never even have to set that one explicitly. Or you could use it to indicate that some kind of init is done.

Reply:
I like this idea. I will see how it works out. Working on updating this PR today.

Reply:
I liked the idea of these general methods, but I found there were only two cases where checking for multiple states, so instead I added `active` and `loaded` computed properties in the `enum`. They are checking the `rawValue` as suggested.

Reply:
> I liked the idea of these general methods, but I found there were only two cases where checking for multiple states, so instead I added active and loaded computed properties in the enum.

Sounds fine to me. But you should name them `isActive` & `isLoaded` which is the naming convention for booleans in most projects. You probably were following `mainWindow.loaded`, but that's a special case...it looks it was created as a substitute for `isLoaded` which already exists but has unwanted side effects.

Reply:
I first coded them as `isActive` and `isLoaded` because I too am used to that convention. I renamed them because I didn't see that convention consistently used. Happy to change them back. Lets see if @uiryuu has to say about IINA conventions.

Reply:
I had a chat with @uiryuu about boolean flag naming conventions. The "is" prefix convention is not preferred. The flags you see in IINA that convention were likely added by me. I was used to that convention from other projects.